components/Elevation/src/UIColor+MaterialElevation.h

```diff
@@ -0,0 +1,40 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <UIKit/UIKit.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ Provides extension to UIColor for Material Elevation usage.
+ */
+@interface UIColor (MaterialElevation)
+
+/**
+ Returns a color that takes the specified elevation value into account.
+ @param elevation The elevation to use when resolving the color.
+ */
+- (UIColor *)mdc_resolvedColorWithElevation:(CGFloat)elevation;
+
+/**
+ Returns a color that takes the specified elevation value and traits into account.
+ @param traitCollection The traits to use when resolving the color.
+ @param elevation The elevation to use when resolving the color.
+ */
+- (UIColor *)mdc_resolvedColorWithTraitCollection:(UITraitCollection *)traitCollection
+                                        elevation:(CGFloat)elevation API_AVAILABLE(ios(13.0));
```

Comment:
If we *really* don't support iOS 13, at a minimum the method should do nothing. At best, it should raise an NSException.

Reply:
Because `resolvedColorWithTraitCollection` and all traitCollection/userStyle was introduced after iOS 13, I don't feel there is a need to make this method support pre 13 and have behavior that users don't understand. If users need to get a resolvedColor for pre-iOS13, they can use the other method provided explicitly.

Reply:
Ok, if we don't want this code executing on iOS 13 (I still think we should allow it), please have it raise an NSException rather than performing undocumented behavior.

Reply:
Addressed 7546111a062e71753a2bd9f5d79ddba71a187cca. Opened the availability to pre iOS 13.